src/StreamJsonRpc/Reflection/RpcMarshalableAttribute.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc;
+
+using System;
+
+/// <summary>
+/// Marks an interface as marshalable.
+/// </summary>
+[AttributeUsage(AttributeTargets.Interface, AllowMultiple = false, Inherited = false)]
+public class RpcMarshalableAttribute : Attribute
```

Comment:
@AArnott, should we have `DisposeTargetOnDisconnect` as a property of this attribute? If we do, should the default be `true` or `false`?

Reply:
I don't think we should. The only behavior should be `true`. My thinking is that the underlying protocol already describes that disposing the proxy _will_ dispose the marshaled object. This means that passing a marshalable object over RPC signifies an ownership transfer. A dropped connection should not leave any resources unfreed, so we should guarantee that Dispose is called on behalf of the proxy if the proxy cannot do it do to a dropped connection.